libs/ui-toolkit/src/components/transaction-dialog/transaction-dialog.tsx

```diff
@@ -0,0 +1,129 @@
+import { t } from '@vegaprotocol/react-helpers';
+import { useEffect, useRef, useState } from 'react';
+import { isEthereumError, TxState } from '@vegaprotocol/react-helpers';
+import { ConfirmRow, TxRow, ConfirmationEventRow } from './dialog-rows';
+import { DialogWrapper } from './dialog-wrapper';
+import { Loader } from '../loader';
+import { Intent } from '../../utils/intent';
+import { Dialog } from '../dialog';
+import { Icon } from '../icon';
+
+export interface TransactionDialogProps<TEvent = undefined> {
+  name: string;
+  status: TxState;
+  error: Error | null;
+  confirmations: number;
+  txHash: string | null;
+  requiredConfirmations?: number;
+  // Undefined means this dialog isn't expecting an additional event for a complete state, null means
+  // it is but hasn't been received yet
+  confirmationEvent?: TEvent | null;
```

Comment:
chore (non-blocking): seems convoluted can we enum this?

Reply:
I'm going to just update this prop to be a `confirmed` boolean. My initial thinking was that we might want to display data from the confirmation event in the dialog (but for all types of events, hence the generic) right now though its only used as essentially a boolean on whether or not to show the final 'confirming...' row